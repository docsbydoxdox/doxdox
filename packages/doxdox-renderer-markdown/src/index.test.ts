import renderer from './index';

describe('markdown', () => {
    beforeAll(() => {
        global.Date = class extends Date {
            toDateString() {
                return 'Mock Date';
            }

            toTimeString() {
                return 'Mock Time';
            }
        } as DateConstructor;
    });

    it('render', async () => {
        expect(
            await renderer({
                name: 'doxdox-example',
                description: 'doxdox example description',
                version: '1.0.0',
                files: [
                    {
                        path: 'directory/index.js',
                        methods: [
                            {
                                slug: 'index-js-methodname',
                                name: 'methodName',
                                fullName: 'methodName(param)',
                                description: 'Method description',
                                params: [
                                    {
                                        name: 'param',
                                        description: 'Param description',
                                        types: ['string']
                                    }
                                ],
                                returns: [
                                    {
                                        name: null,
                                        description: 'Return description',
                                        types: ['void']
                                    }
                                ],
                                private: false
                            }
                        ]
                    }
                ]
            })
        ).toMatchSnapshot();
    });
});
