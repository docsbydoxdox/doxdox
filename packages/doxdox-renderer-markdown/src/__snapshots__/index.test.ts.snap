// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`markdown render 1`] = `
"# doxdox-example

> doxdox example description

## methodName(param)

Method description

### Parameters

| Name  | Types  | Description       |
| ----- | ------ | ----------------- |
| param | string | Param description |

### Returns

void
Return description

Documentation generated with [doxdox](https://github.com/docsbydoxdox/doxdox)

Generated on Mock Date Mock Time
"
`;
